### PR TITLE
Add image context to viewer configuration

### PR DESCRIPTION
**Pull request by chrisvanrun**
For viewers to better adapt to different medical imaging modalities it is now possible to set an image context for a viewer configuration. 

https://user-images.githubusercontent.com/7871310/109066897-0e59eb00-76ee-11eb-97df-b5e906b6db54.mp4

Note: OBS somehow didn't pick up the dropdown menu: it's there! Current image-context options are:
* Pathology
* Ophthalmology
* Multiparametric MRI

Relates to https://github.com/DIAGNijmegen/rse-cirrus-core/issues/992

### CODE REVIEW

**Comment by jmsmkn** [comment]
This has the changes from another PR included, please could you rebase/cherry pick only the intended changes for this feature?

**Comment by chrisvanrun** [comment]
Accidently branched this of another branch instead of master. Fixed via a forced push since no-one reviewed this yet.

**Comment by HarmvZ** [comment]
Thanks for picking this up Chris! One quick comment (not an full review): You have a typo in op**h**thalmology.

File: app/grandchallenge/workstation_configs/models.py [hunk 1]
```diff
@@ -144,8 +144,21 @@ class WorkstationConfig(TitleSlugDescriptionModel, UUIDModel):
         (SLAB_RENDER_METHOD_AVERAGE, "Average"),
     )
 
+    IMAGE_CONTEXT_DEFAULT = "DEF"
+    IMAGE_CONTEXT_PATHOLOGY = "PATH"
+    IMAGE_CONTEXT_OPTHAMOLOGY = "OPTHA"
```
Comment (jmsmkn) [inline app/grandchallenge/workstation_configs/models.py:149]: Please correct the spelling in both the strings and code.

File: app/grandchallenge/workstation_configs/models.py [hunk 2]
```diff
@@ -144,8 +144,21 @@ class WorkstationConfig(TitleSlugDescriptionModel, UUIDModel):
...
+    IMAGE_CONTEXT_MPMRI = "MPMRI"
+
+    IMAGE_CONTEXT_CHOICES = (
+        (IMAGE_CONTEXT_DEFAULT, "Default"),
+        (IMAGE_CONTEXT_PATHOLOGY, "Pathology"),
+        (IMAGE_CONTEXT_OPTHAMOLOGY, "Opthamology"),
+        (IMAGE_CONTEXT_MPMRI, "Multiparametric MRI"),
+    )
```
Comment (jmsmkn) [inline app/grandchallenge/workstation_configs/models.py:157]: Django has a nicer way of defining these now with `TextChoices` or `IntegerChoices`. You can see it used here https://github.com/comic/grand-challenge.org/blob/51a2392a137411e8f7d84c0c6b5b7dc87c9f0041/app/grandchallenge/workstations/models.py#L282-L309

Documentation here https://docs.djangoproject.com/en/3.1/ref/models/fields/#enumeration-types

This is preferred for new choice options, the old stuff is being refactored whenever I touch it.

File: app/grandchallenge/workstation_configs/templates/workstation_configs/workstationconfig_detail.html [hunk 3]
```diff
@@ -25,6 +25,10 @@ <h1>Viewer Configuration {{ object.title }}</h1>
     {% endif %}
 
     <table class="table table-hover my-3">
+           <tr>
```
Comment (jmsmkn) [inline app/grandchallenge/workstation_configs/templates/workstation_configs/workstationconfig_detail.html:28]: Nitpick - please tidy your templates, the PyCharm autoformatter is pretty good.

File: app/grandchallenge/workstation_configs/models.py [hunk 4]
```diff
@@ -168,6 +182,13 @@ class WorkstationConfig(TitleSlugDescriptionModel, UUIDModel):
         related_name="workstation_default_window_presets",
     )
 
+    image_context = models.CharField(
+        blank=True,
+        max_length=6,
+        default=IMAGE_CONTEXT_DEFAULT,
```
Comment (jmsmkn) [inline app/grandchallenge/workstation_configs/models.py:188]: It's weird to have a "default" choice item and let this model field be blank. What is the different between "default" and "blank"? If there is no difference, remove `IMAGE_CONTEXT_DEFAULT` as a choice and do not set a default in the model.

**Review by jmsmkn — CHANGES_REQUESTED** [review]

File: app/grandchallenge/workstation_configs/models.py (hunk 4 above)
Comment (chrisvanrun) [reply app/grandchallenge/workstation_configs/models.py:188]: I had default initially, but it interfered with some testing and it felt weird having a required field. In the end, I picked Harm's initial instinct to add a Default to the enum. However, I do agree a do real blank is nicer.

File: app/grandchallenge/workstation_configs/templates/workstation_configs/workstationconfig_detail.html (hunk 3 above)
Comment (chrisvanrun) [reply app/grandchallenge/workstation_configs/templates/workstation_configs/workstationconfig_detail.html:28]: Going to double-check my PyCharm settings, it should have picked this up.

File: app/grandchallenge/workstation_configs/models.py (hunk 2 above)
Comment (chrisvanrun) [reply app/grandchallenge/workstation_configs/models.py:157]: Came across this, but thought I'd keep everything in the style that was chosen. Will refactor.

**Comment by chrisvanrun** [comment]
> 
> 
> Thanks for picking this up Chris! One quick comment (not an full review): You have a typo in op**h**thalmology.

Of course! The good old 'ph' going for an _/f/_ phoneme. Thanks for the check!

File: app/grandchallenge/workstation_configs/models.py [hunk 5]
```diff
@@ -124,32 +124,24 @@
...
+        AVERAGE = "AVG", "Average"
 
-    SLAB_RENDER_METHOD_CHOICES = (
-        (SLAB_RENDER_METHOD_MAXIMUM, "Maximum"),
-        (SLAB_RENDER_METHOD_MINIMUM, "Minimum"),
-        (SLAB_RENDER_METHOD_AVERAGE, "Average"),
-    )
+    class ImageContext(models.TextChoices):
+        PATHOLOGY = "PATH", "Pathology"
+        OPHTHAMOLOGY = "OPHTH", "Ophthamology"
```
Comment (jmsmkn) [inline app/grandchallenge/workstation_configs/models.py:139]: Missing an `l` now, it's `OPHTHALMOLOGY`/Ophthalmology

**Review by jmsmkn — CHANGES_REQUESTED** [review]
Getting there :-)

**Comment by chrisvanrun** [comment]
I think '_supercalifragilisticexpialidocious_' is easier to spell than '_ophthalmology_'. 

In my defense =D: https://trends.google.nl/trends/explore?date=all&q=Ophthalmology,Ophthamology,Opthalmology

File: app/grandchallenge/workstation_configs/models.py (hunk 5 above)
Comment (chrisvanrun) [reply app/grandchallenge/workstation_configs/models.py]: Either Github, or I am confused. But this seems fixed but is still blocking. So I am going to dismiss the review.

**Review by MikeOverkamp-diag — APPROVED** [review]

**Pull request merged**